src/utils/eth_transaction.cairo

```diff
@@ -125,6 +133,13 @@ namespace EthTransaction {
         );
     }
 
+    // @notice Decode a modern Ethereum transaction
+    // @dev This function decodes a modern Ethereum transaction in accordance with EIP-1559.
```

Comment:
actually is not EIP1559 (tx type 2) but before, as you mentioned later EIP-2718. This also supports tx type 1 (access list added)

Reply:
Updated